src/main/java/dev/blachut/svelte/lang/parsing/html/SvelteElementResolveScopeProvider.kt

```diff
@@ -0,0 +1,35 @@
+package dev.blachut.svelte.lang.parsing.html
+
+import com.intellij.lang.javascript.DialectDetector
+import com.intellij.lang.javascript.ecmascript6.TypeScriptResolveScopeProvider
+import com.intellij.lang.javascript.psi.resolve.JSElementResolveScopeProvider
+import com.intellij.lang.javascript.psi.util.JSUtils
+import com.intellij.lang.typescript.library.TypeScriptLibraryProvider
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiElement
+import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.psi.xml.XmlFile
+import dev.blachut.svelte.lang.SvelteHtmlFileType
+
+class SvelteElementResolveScopeProvider : JSElementResolveScopeProvider {
+    private val tsProvider = object : TypeScriptResolveScopeProvider() {
+        override fun isApplicable(file: VirtualFile) = true
+
+        override fun restrictByFileType(file: VirtualFile, libraryService: TypeScriptLibraryProvider, moduleAndLibraryScope: GlobalSearchScope): GlobalSearchScope {
+            return super.restrictByFileType(file, libraryService, moduleAndLibraryScope)
+                .uniteWith(GlobalSearchScope.getScopeRestrictedByFileTypes(moduleAndLibraryScope, file.fileType))
+        }
+    }
+
+    override fun getElementResolveScope(element: PsiElement): GlobalSearchScope? {
+        val psiFile = element.containingFile
+        if (psiFile?.fileType !is SvelteHtmlFileType) return null
+        if (psiFile !is XmlFile) return null
+        val scriptTag = JSUtils.findScriptTagContent(psiFile)
```

Comment:
Svelte can have two script tags and they could have different languages, this finds first one, is it okay?

Reply:
I think we can live with it as is. I will try to find a better strategy but I am not sure

Reply:
We way we handle this in the language server is to check script and then module script and pick the first lang/type tag we find, so if someone chooses TS in the script tag he will automatically use it in the other one as well - no language mixes.

Reply:
Seems like a good balance 👍